torchao/prototype/mx_formats/mx_ops.py

```diff
@@ -52,72 +57,137 @@ def decorator(func):
     return decorator
 
 
-@implements([aten.detach.default])
-def mx_desugar_op(aten_op, args, kwargs=None):
-    old = args[0]
-    new_data = aten_op(old._data, *args[1:], **kwargs)
-    new = MXTensor(
-        old._scale_e8m0,
-        new_data,
-        old._elem_dtype,
-        old._block_size,
-        old._orig_dtype,
-        old._use_fp4_custom_triton_dequant_kernel,
-        old._gemm_kernel_choice,
-        old._pack_fp6,
+def implements_func(torch_ops):
```

Comment:
IMO it would be cleaner to keep `__torch_function__` and `__torch_dispatch__` functions in separate files to reduce confusion